### DISCUSSION TITLE
DevelopmentConstant and group by.

### DISCUSSION THREAD

**Discussion by MathiasOstnes** [casact/chainladder-python#467] [Q&A]
Hi, i am working on a larger model for claims triangles. I have the problem that i have chosen development constant from an excel file that looks like this: 

LOB | DevelopmentQuarter | Value
Type1 | 3 | 1.5344
Type2| 3| 1.234

How does the groupby within the DevelopmentConstant work in practice? When i try to groupby class_path it gives me an error. 

A more general question i have is if we are given the development factors in a excel file, what would be the most efficient way to implement them in the DevelopmentConstant function?

Thanks for the documentation already, and thanks for all the hard work,

**Comment by jbogaardt** [comment]
I do something similar in practice.

The `DevelopmentConstant` can take a callable (i.e. function) that works similar to `pd.apply`. A good example is in [the docs](https://chainladder-python.readthedocs.io/en/latest/gallery/plot_callable_dev_constant.html). 

The example uses a dictionary, but I think the interchange between Excel and Pandas is pretty straightforward and the interchange between a pandas dataframe and a dictionary is also pretty straight forward.
